promdump/promdump.go

```diff
@@ -651,9 +770,52 @@ func main() {
 		}
 	}
 	if *metric != "" {
-		err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
+		customMetricCount, err = exportMetric(ctx, promApi, *metric, beginTS, endTS, *periodDur, *batchDur, *batchesPerFile, *out)
 		if err != nil {
 			log.Fatalln("exportMetric:", err)
 		}
 	}
+
+	if *enableTar {
+		var filename string
+		filename = *tarFilename
+		if *tarFilename == "" {
+			// If filename is not provided, generate a default tar filename
+			filename = generateDefaultTarFilename()
+		} else {
+			// Check if filename already exists
+			if _, err := os.Stat(filename); err == nil {
+				log.Fatalf("Error: %s already exists. Please choose a different filename.\n", filename)
+				return
+			}
+		}
+
+		out, err := os.Create(filename)
+		if err != nil {
+			log.Printf("Error writing archive: %v (File: %s)\n", err, filename)
+			return
+		}
+
+		// Create the archive
+		err = createArchive(out)
+		if err != nil {
+			log.Printf("Error creating archive: %v\n", err)
+		}
+		// cleaning files
+		for _, v := range collectMetrics {
+			metricName, err := getMetricName(v)
+			if err != nil {
+				log.Fatalf("Error removing files: %v", err)
+			}
+			if v.collect {
+				_, err := cleanFiles(metricName, v.fileCount)
+				if err != nil {
+					log.Fatalf("Error removing files: %v", err)
+				}
+			}
+		}
+
+		fmt.Println("Finished creating metrics Tarball file:", filename)
```

Comment:
It's customarily "tarball" or "tar file" rather than "tarball file". Tarball is pretty informal, so I'd go with "tar file" or "metrics bundle" or similar.

Reply:
yes I agree Tarball file don't suit properly, I used the metrics bundle instead